platform/packages/finance/src/coin/dto/amount_serde.rs

```diff
@@ -0,0 +1,105 @@
+use std::fmt::Formatter;
+
+/// This module implements (de-)serialation of Amount as String
+/// to keep compatibility with pre-CW 2.x
+use serde::{
+    de::{Unexpected, Visitor},
+    Deserializer, Serializer,
+};
+
+use crate::coin::Amount;
+
+pub(super) fn serialize<S>(amount: &Amount, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    serializer.serialize_str(&amount.to_string())
+}
+
+pub(super) fn deserialize<'de, D>(deserializer: D) -> Result<Amount, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    struct StrVisitor();
+    impl<'de> Visitor<'de> for StrVisitor {
+        type Value = Amount;
+
+        fn expecting(&self, formatter: &mut Formatter) -> std::fmt::Result {
+            formatter.write_str("\"<u128>\"")
+        }
+
+        fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
+        where
+            E: serde::de::Error,
+        {
+            str::parse(v).map_err(|_| E::invalid_value(Unexpected::Str(v), &self))
```

Comment:
[Style] Why not use the method directly?
```suggestion
            v.parse().map_err(|_| E::invalid_value(Unexpected::Str(v), &self))
```